tests/lib/parse.js

```diff
@@ -82,4 +82,24 @@ describe("parse()", () => {
         });
 
     });
+
+    describe("acorn", () => {
+        it("acorn.parse() should not re-use the same identifier node in object patterns", () => {
+            const code = "({x, y = 1} = {z})";
+
+            const ast = espree.parse(code, { ecmaVersion: 2015 });
+
+            const assignment = ast.body[0].expression;
+            const objectPattern = assignment.left;
+            const objectExpression = assignment.right;
+
+            const propertyX = objectPattern.properties[0];
+            const propertyY = objectPattern.properties[1];
+            const propertyZ = objectExpression.properties[0];
+
+            assert.notStrictEqual(propertyX.key, propertyX.value); // true in v7, false in v8
+            assert.notStrictEqual(propertyY.key, propertyY.value.left); // true in v7, false in v8
+            assert.notStrictEqual(propertyZ.key, propertyZ.value); // true in v7, false in v8
```

Comment:
```suggestion
            assert.notStrictEqual(propertyX.key, propertyX.value);
            assert.notStrictEqual(propertyY.key, propertyY.value.left);
            assert.notStrictEqual(propertyZ.key, propertyZ.value);
```

I think there's no need for the version comments here.